plugins/generate-proptypes/index.mjs

```diff
@@ -0,0 +1,92 @@
+import * as tsUtils from "./ts-utils.mjs";
+import injectPropTypes from "./inject-prop-types.mjs";
+
+/**
+ * Custom Babel plugin that injects `import PropTypes from "prop-types";`
+ * and js proptypes in the format `Component.propTypes = {...};` into
+ * TypeScript components just before they are compiled to JavaScript.
+ */
+
+export default async function ({ types: t }) {
+  const tsProgram = await tsUtils.generateTsProgram();
+
+  return {
+    visitor: {
+      ImportDeclaration(path, state) {
+        const { filename } = state;
+        if (filename.includes(".component.tsx")) {
+          // Find React import and insert prop types after
+          if (t.isStringLiteral(path.node.source, { value: "react" })) {
+            path.insertAfter(
+              t.importDeclaration(
+                [t.importDefaultSpecifier(t.identifier("PropTypes"))],
+                t.stringLiteral("prop-types")
+              )
+            );
+          }
+        }
+      },
+      FunctionDeclaration(path, state) {
+        function inject(props, nodeName) {
```

Comment:
Can this function be moved to a shared scope so it's not defined here and in `VariableDeclarator` if you add a `path` param as it seems like the only difference is one uses the `path` and the other `parentPath`?

```
 function inject(props, nodeName, path) {
          injectPropTypes(t, {
            path,
            props,
            nodeName,
          });
        }
```